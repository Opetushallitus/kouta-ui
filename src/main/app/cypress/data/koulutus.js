import { merge } from 'lodash';

const getBaseFields = () => {
  return {
    oid: '1.2.246.562.13.00000000000000000072',
    koulutustyyppi: 'amm',
    koulutusKoodiUri: 'koulutus_0#1',
    tila: 'tallennettu',
    tarjoajat: ['1.2.246.562.10.81934895871'],
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: {
      tyyppi: 'amm',
      kuvaus: {},
      lisatiedot: [
        {
          otsikkoKoodiUri: 'koulutuksenlisatiedot_0#1',
          teksti: {
            fi: 'koulutuksenlisatiedot_0 kuvaus',
          },
        },
      ],
    },
    julkinen: true,
    muokkaaja: '1.2.246.562.24.62301161440',
    organisaatioOid: '1.2.246.562.10.594252633210',
    kielivalinta: ['fi', 'sv'],
    modified: '2019-04-01T13:01',
  };
};

const getAmmatillinenFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    koulutustyyppi: tyyppi,
    koulutusKoodiUri: 'koulutus_0#1',
    tila: 'tallennettu',
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: { tyyppi },
  });
};

const getKorkeakouluFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    johtaaTutkintoon: true,
    koulutustyyppi: tyyppi,
    koulutusKoodiUri: 'koulutus_0#1',
    nimi: { fi: 'Fi nimi', sv: 'Sv nimi' },
    metadata: {
      tyyppi,
      kuvaus: { fi: 'Fi kuvaus', sv: 'Sv kuvaus' },
      tutkintonimikeKoodiUrit: ['tutkintonimikekk_1#1', 'tutkintonimikekk_2#1'],
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_1#1',
      kuvauksenNimi: { fi: 'Fi kuvauksen nimi', sv: 'Sv kuvauksen nimi' },
      koulutusalaKoodiUrit: [
        'kansallinenkoulutusluokitus2016koulutusalataso2_1#1',
      ],
    },
  });
};

const getLukioFields = ({ tyyppi }) => getAmmatillinenFields({ tyyppi });

export default ({ tyyppi = 'amm' } = {}) => {
  if (tyyppi === 'amm') {
    return getAmmatillinenFields({ tyyppi });
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return getKorkeakouluFields({ tyyppi });
  } else if (tyyppi === 'lk') {
    return getLukioFields({ tyyppi });
  }

  return getBaseFields();
};
