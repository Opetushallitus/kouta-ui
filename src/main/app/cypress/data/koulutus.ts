import { merge } from 'lodash/fp';

const getBaseFields = ({ tyyppi = 'amm' } = {}) => {
  return {
    oid: '1.2.246.562.13.00000000000000000072',
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_0#1'],
    tila: 'tallennettu',
    tarjoajat: ['1.2.246.562.10.81934895871'],
    nimi: {
      en: 'Vocational qualification in Agriculture',
      fi: 'Maatalousalan perustutkinto',
      sv: 'Grundexamen inom lantbruksbranschen',
    },
    metadata: {
      tyyppi: tyyppi,
      kuvaus: {},
      lisatiedot: [
        {
          otsikkoKoodiUri: 'koulutuksenlisatiedot_0#1',
          teksti: {
            fi: 'koulutuksenlisatiedot_0 kuvaus',
          },
        },
      ],
    },
    julkinen: true,
    muokkaaja: '1.2.246.562.24.62301161440',
    organisaatioOid: '1.2.246.562.10.594252633210',
    kielivalinta: ['fi', 'sv'],
    modified: '2019-04-01T13:01',
  };
};

const getAmmatillinenFields = ({ tyyppi = 'amm' }) => {
  return merge(getBaseFields(), {
    oid: '1.2.246.562.13.00000000000000000599',
    johtaaTutkintoon: false,
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_351107#12'],
    tila: 'tallennettu',
    tarjoajat: [],
    nimi: {
      en: 'Vocational qualification in Mining',
      fi: 'Kaivosalan perustutkinto',
      sv: 'Grundexamen inom gruvbranschen',
    },
    metadata: {
      tyyppi: tyyppi,
      kuvaus: {},
      lisatiedot: [],
    },
    julkinen: true,
    esikatselu: true,
    muokkaaja: '1.2.246.562.24.26603962037',
    organisaatioOid: '1.2.246.562.10.60198812368',
    kielivalinta: ['fi', 'sv'],
    ePerusteId: 6777660,
    modified: '2020-09-21T16:27',
  });
};

const getKorkeakouluFields = ({ tyyppi }) => {
  return merge(getBaseFields(), {
    johtaaTutkintoon: true,
    koulutustyyppi: tyyppi,
    koulutuksetKoodiUri: ['koulutus_671112#12'],
    nimi: { fi: 'Fysioterapeutti (AMK)', sv: 'Fysioterapeut (YH)' },
    metadata: {
      tyyppi,
      kuvaus: { fi: 'Fi kuvaus', sv: 'Sv kuvaus' },
      tutkintonimikeKoodiUrit: ['tutkintonimikekk_1#1', 'tutkintonimikekk_2#1'],
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_1#1',
      kuvauksenNimi: { fi: 'Fi kuvauksen nimi', sv: 'Sv kuvauksen nimi' },
      koulutusalaKoodiUrit: [
        'kansallinenkoulutusluokitus2016koulutusalataso2_052#1',
      ],
    },
  });
};

const getLukioFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    koulutuksetKoodiUri: ['koulutus_309902#7'],
    nimi: { fi: 'Lukion oppimäärä' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_150',
    },
  });
};

const getTuvaFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    nimi: { fi: 'Tutkintokoulutukseen valmentava koulutus (TUVA)' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_38',
      kuvaus: { fi: 'kuvausteksti' },
    },
  });
};

const getTelmaFields = ({ tyyppi }) => {
  return merge(getBaseFields({ tyyppi }), {
    nimi: { fi: 'Työhön ja itsenäiseen elämään valmentava koulutus (TELMA)' },
    metadata: {
      opintojenLaajuusKoodiUri: 'opintojenlaajuus_60',
      kuvaus: { fi: 'kuvausteksti' },
    },
  });
};

export default ({ tyyppi = 'amm' } = {}) => {
  if (tyyppi.startsWith('amm')) {
    return getAmmatillinenFields({ tyyppi });
  } else if (['yo', 'amk'].includes(tyyppi)) {
    return getKorkeakouluFields({ tyyppi });
  } else if (tyyppi === 'lk') {
    return getLukioFields({ tyyppi });
  } else if (tyyppi === 'tuva') {
    return getTuvaFields({ tyyppi });
  } else if (tyyppi === 'telma') {
    return getTelmaFields({ tyyppi });
  }

  return getBaseFields();
};
