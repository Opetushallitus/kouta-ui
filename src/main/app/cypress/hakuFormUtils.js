import merge from 'lodash/merge';

import organisaatio from './data/organisaatio';
import {
  stubKoodistoRoute,
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
} from './utils';

export const stubHakuFormRoutes = ({ cy, organisaatioOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  stubKoodistoRoute({ koodisto: 'haunkohdejoukko', cy });
  stubKoodistoRoute({ koodisto: 'hakutapa', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });
  stubKoodistoRoute({ koodisto: 'opetuspaikkakk', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });

  cy.route({
    method: 'GET',
    url: '**/haku/list**',
    response: [],
  });

  stubHakemuspalveluLomakkeetRoute({ cy });
  stubOppijanumerorekisteriHenkiloRoute({ cy });
};
