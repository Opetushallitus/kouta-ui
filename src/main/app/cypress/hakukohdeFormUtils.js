import merge from 'lodash/merge';

import { stubKoodistoRoute } from './utils';
import organisaatio from './data/organisaatio';
import haku from './data/haku';

export const stubHakukohdeFormRoutes = ({ cy, organisaatioOid, hakuOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  stubKoodistoRoute({ koodisto: 'pohjakoulutusvaatimustoinenaste', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });
  stubKoodistoRoute({ koodisto: 'valintakokeentyyppi', cy });
  stubKoodistoRoute({ koodisto: 'liitetyypitamm', cy });

  cy.route({
    method: 'GET',
    url: `**/haku/${hakuOid}`,
    response: merge(haku(), {
      oid: hakuOid,
      organisaatioOid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'GET',
    url: `**/hakukohde/list**`,
    response: [],
  });
};
