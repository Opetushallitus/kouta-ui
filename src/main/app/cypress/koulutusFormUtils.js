import merge from 'lodash/merge';

import { stubKoodistoRoute } from './utils';
import organisaatioHierarkia from './data/organisaatioHierarkia';
import organisaatio from './data/organisaatio';
import koodisto from './data/koodisto';

export const stubKoulutusFormRoutes = ({ cy, organisaatioOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae?oid=${organisaatioOid}**`,
    response: organisaatioHierarkia({ rootOid: organisaatioOid }),
  });

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-ylakoodit/koulutustyyppi_*',
    response: koodisto({ koodisto: 'koulutus' }),
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/list?organisaatioOid=${organisaatioOid}`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_0',
    response: [
      {
        koodiUri: 'koulutus_0',
        versio: 1,
        metadata: [{ kieli: 'fi', nimi: 'Nimi' }],
      },
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/eperusteet-service/api/perusteet?tuleva=true&siirtyma=false&voimassaolo=true&poistunut=false&kieli=fi&koulutuskoodi=koulutus_0',
    response: {
      data: [
        {
          kuvaus: { fi: 'koulutus_0 kuvaus' },
          osaamisalat: [],
          tutkintonimikeKoodiUri: 'nimike_1#1',
          id: '1',
        },
      ],
    },
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_0**',
    response: [
      {
        koodiUri: 'okmohjauksenala_1',
        metadata: [{ kieli: 'fi', nimi: 'Ala' }],
      },
      {
        koodiUri: 'opintojenlaajuus_1',
        metadata: [{ kieli: 'fi', nimi: '180' }],
      },
      {
        koodiUri: 'opintojenlaajuusyksikko_1',
        metadata: [{ kieli: 'fi', nimi: 'op' }],
      },
    ],
  });

  stubKoodistoRoute({ koodisto: 'tutkintonimikekk', cy });
  stubKoodistoRoute({ koodisto: 'opintojenlaajuus', cy });
  stubKoodistoRoute({ koodisto: 'koulutuksenjarjestamisenlisaosiot', cy });
};
