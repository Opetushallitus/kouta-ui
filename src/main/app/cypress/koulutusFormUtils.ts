import { merge } from 'lodash';

import {
  stubKoodistoRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubEPerusteetByKoulutuskoodiRoute,
  stubCommonRoutes,
} from '#/cypress/utils';
import organisaatioHierarkia from '#/cypress/data/organisaatioHierarkia';
import organisaatio from '#/cypress/data/organisaatio';
import koodisto from '#/cypress/data/koodisto';
import koodistoKoulutus351107 from './data/koodistoKoulutus351107';
import koodistoKoulutukset from './data/koodistoKoulutukset';
import koodistoKoulutus351107Alakoodit from './data/koodistoKoulutus351107Alakoodit';
import koodistoKoulutus309902 from './data/koodistoKoulutus309902';
import koodistoKoulutus309902Alakoodit from './data/koodistoKoulutus309902Alakoodit';
import koodistoKoulutus671112 from './data/koodistoKoulutus671112';
import koodistoKoulutus671112Alakoodit from './data/koodistoKoulutus671112Alakoodit';
import ePerustePerusteenosat6777674 from './data/ePerustePerusteenosat6777674';
import ePerustePerusteenosat6778360 from './data/ePerustePerusteenosat6778360';
import ePeruste6777660TutkinnonOsat from './data/ePeruste6777660TutkinnonOsat';
import ePeruste6777660Rakenne from './data/ePeruste6777660Rakenne';
import ePeruste6777660 from './data/ePeruste6777660';

export const stubKoulutusFormRoutes = ({ organisaatioOid }) => {
  stubCommonRoutes();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae**`,
    response: organisaatioHierarkia({ rootOid: organisaatioOid }),
  });

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-ylakoodit/koulutustyyppi_*',
    response: koodisto({ koodisto: 'koulutus' }),
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/list?organisaatioOid=${organisaatioOid}`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/*/toteutukset/list**`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_351107/**',
    response: koodistoKoulutus351107,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_351107**',
    response: koodistoKoulutus351107Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_309902/**',
    response: koodistoKoulutus309902,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_309902**',
    response: koodistoKoulutus309902Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_671112/**',
    response: koodistoKoulutus671112,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_671112**',
    response: koodistoKoulutus671112Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/json/koulutus/koodi?onlyValidKoodis=true**',
    response: koodistoKoulutukset,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6777674',
    response: ePerustePerusteenosat6777674,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6778360',
    response: ePerustePerusteenosat6778360,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6778360',
    response: ePerustePerusteenosat6778360,
  });

  cy.route({
    method: 'GET',
    url: '**/tutkinnonosat',
    response: ePeruste6777660TutkinnonOsat,
  });

  cy.route({
    method: 'GET',
    url:
      '**/eperusteet-service/api/perusteet/6777660/suoritustavat/reformi/rakenne',
    response: ePeruste6777660Rakenne,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteet/6777660',
    response: ePeruste6777660,
  });

  stubEPerusteetByKoulutuskoodiRoute();

  stubKoodistoRoute({ koodisto: 'tutkintonimikekk' });
  stubKoodistoRoute({ koodisto: 'opintojenlaajuus' });
  stubKoodistoRoute({ koodisto: 'koulutuksenlisatiedot' });

  stubKoodistoRoute({
    koodisto: 'kansallinenkoulutusluokitus2016koulutusalataso2',
  });

  stubOppijanumerorekisteriHenkiloRoute();
};
