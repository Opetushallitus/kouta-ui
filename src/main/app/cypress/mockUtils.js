const path = require('path');

const MOCK_PATH = 'cypress/mocks';

export const addMockFileRoutes = fileName => {
  cy.readFile(path.join(MOCK_PATH, fileName)).then(data => {
    data.forEach(({ url, method = 'GET', response }) => {
      // TODO: Mock only on matching response body and use cy.intercept()
      cy.route({
        method,
        url,
        status: response.status,
        response: response.body,
      });
    });
  });
};
