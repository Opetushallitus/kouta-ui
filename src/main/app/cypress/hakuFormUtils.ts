import { playMocks } from 'kto-ui-common/cypress/mockUtils';
import { merge } from 'lodash/fp';

import organisaatio from '#/cypress/data/organisaatio';
import organisaatioHierarkia from '#/cypress/data/organisaatioHierarkia';
import hakuMocks from '#/cypress/mocks/haku.mocks.json';

import {
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubCommonRoutes,
} from './utils';

export const stubHakuFormRoutes = ({ organisaatioOid }) => {
  stubCommonRoutes();
  playMocks(hakuMocks);

  cy.intercept(
    {
      method: 'GET',
      url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae**oid=${organisaatioOid}**`,
    },
    { body: organisaatioHierarkia({ rootOid: organisaatioOid }) }
  );

  cy.intercept(
    {
      method: 'GET',
      url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae**`,
    },
    { body: organisaatioHierarkia({ rootOid: organisaatioOid }) }
  );

  cy.intercept(
    {
      method: 'GET',
      url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    },
    {
      body: merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    }
  );

  cy.intercept(
    {
      method: 'POST',
      url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    },
    {
      body: [
        merge(organisaatio(), {
          oid: organisaatioOid,
        }),
      ],
    }
  );

  cy.intercept({ method: 'GET', url: '**/haku/list**' }, { body: [] });

  cy.intercept({ method: 'GET', url: `**/toteutus/list**` }, { body: [] });

  cy.intercept({ method: 'GET', url: `**/koulutus/list**` }, { body: [] });

  cy.intercept({ method: 'GET', url: `**/search/haku/**` }, { body: [] });

  stubHakemuspalveluLomakkeetRoute();
  stubOppijanumerorekisteriHenkiloRoute();
};
