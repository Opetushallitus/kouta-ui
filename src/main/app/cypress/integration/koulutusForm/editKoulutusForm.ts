import { playMocks } from 'kto-ui-common/cypress/mockUtils';
import { merge } from 'lodash/fp';

import koulutus from '#/cypress/data/koulutus';
import { stubKoulutusFormRoutes } from '#/cypress/koulutusFormUtils';
import koulutusMocks from '#/cypress/mocks/koulutus.mocks.json';
import {
  fillKieliversiotSection,
  tallenna,
  assertNoUnsavedChangesDialog,
  getByTestId,
  paste,
  wrapMutationTest,
} from '#/cypress/utils';
import { ENTITY, OPETUSHALLITUS_ORGANISAATIO_OID } from '#/src/constants';

const organisaatioOid = '1.1.1.1.1.1';
const koulutusOid = '1.2.3.4.5.6';

const prepareTest = tyyppi => {
  const testKoulutusFields = {
    oid: koulutusOid,
    organisaatioOid: organisaatioOid,
    tarjoajat: ['1.1.1.1.1.1', '1.2.1.1.1.1'],
  };

  stubKoulutusFormRoutes({ organisaatioOid });

  cy.intercept(
    { method: 'GET', url: `**/koulutus/${koulutusOid}/toteutukset` },
    { body: [] }
  );

  cy.intercept({ method: 'GET', url: `**/toteutus/list**` }, { body: [] });

  cy.intercept({ method: 'GET', url: '**/search/koulutus/**' }, { body: [] });

  cy.intercept(
    { method: 'GET', url: `**/koulutus/${koulutusOid}` },
    { body: merge(koulutus({ tyyppi }), testKoulutusFields) }
  );
};

export const editKoulutusForm = () => {
  beforeEach(() => {
    playMocks(koulutusMocks);
  });

  const mutationTest = wrapMutationTest({
    oid: koulutusOid,
    entity: ENTITY.KOULUTUS,
  });

  it(
    'should be able to edit ammatillinen koulutus',
    mutationTest(() => {
      prepareTest('amm');
      cy.visit(
        `/organisaatio/${organisaatioOid}/koulutus/${koulutusOid}/muokkaus`
      );

      fillKieliversiotSection();

      tallenna();
    })
  );

  it(
    'should be able to edit AMK-koulutus',
    mutationTest(() => {
      prepareTest('amk');
      cy.visit(
        `/organisaatio/${organisaatioOid}/koulutus/${koulutusOid}/muokkaus`
      );

      fillKieliversiotSection();

      tallenna();
    })
  );

  it(
    'should be able to edit lukiokoulutus',
    mutationTest(() => {
      prepareTest('lk');
      cy.visit(
        `/organisaatio/${organisaatioOid}/koulutus/${koulutusOid}/muokkaus`
      );

      fillKieliversiotSection();

      tallenna();
    })
  );

  it('should be able to edit TUVA-koulutus', () => {
    prepareTest('tuva');
    cy.visit(
      `/organisaatio/${organisaatioOid}/koulutus/${koulutusOid}/muokkaus`
    );
    cy.intercept(
      { method: 'POST', url: '**/koulutus' },
      {
        body: {
          muokattu: false,
        },
      }
    ).as('updateTuvaKoulutusResponse');

    getByTestId('linkkiEPerusteisiinInput')
      .find('input')
      .pipe(paste('http://testilinkki.fi'));

    tallenna();

    cy.wait('@updateTuvaKoulutusResponse').then(({ request }) => {
      cy.wrap(request.body).toMatchSnapshot();
    });
  });

  it("Shouldn't complain about unsaved changes for untouched form", () => {
    prepareTest('amm');
    cy.visit(
      `/organisaatio/${organisaatioOid}/koulutus/${koulutusOid}/muokkaus`
    );
    assertNoUnsavedChangesDialog();
  });

  it('Should redirect from url without organization', () => {
    prepareTest('amm');
    cy.visit(`/koulutus/${koulutusOid}/muokkaus`);
    cy.url().should(
      'include',
      `/organisaatio/${OPETUSHALLITUS_ORGANISAATIO_OID}/koulutus/${koulutusOid}/muokkaus`
    );
  });
};
