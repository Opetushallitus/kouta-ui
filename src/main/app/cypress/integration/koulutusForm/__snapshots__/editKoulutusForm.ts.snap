exports[`editKoulutusForm > should be able to edit ammatillinen koulutus #0`] =
{
  "ePerusteId": 1,
  "esikatselu": true,
  "johtaaTutkintoon": true,
  "julkinen": true,
  "kielivalinta": [
    "fi"
  ],
  "koulutusKoodiUri": "koulutus_0#1",
  "koulutustyyppi": "amm",
  "metadata": {
    "koulutusalaKoodiUrit": [],
    "koulutustyyppi": "amm",
    "kuvauksenNimi": {},
    "kuvaus": {},
    "lisatiedot": [
      {
        "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
        "teksti": {
          "fi": "koulutuksenlisatiedot_0 kuvaus"
        }
      }
    ],
    "opintojenLaajuusKoodiUri": null,
    "osaamisalaKoodiUri": null,
    "tutkinnonOsat": [],
    "tutkintonimikeKoodiUrit": [],
    "tyyppi": "amm"
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "koulutus_0"
  },
  "oid": "1.2.3.4.5.6",
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "1.1.1.1.1.1",
    "1.2.1.1.1.1"
  ],
  "teemakuva": null,
  "tila": "tallennettu"
};

exports[`editKoulutusForm > should be able to edit korkeakoulu koulutus #0`] =
{
  "ePerusteId": null,
  "esikatselu": true,
  "johtaaTutkintoon": true,
  "julkinen": true,
  "kielivalinta": [
    "fi"
  ],
  "koulutusKoodiUri": "koulutus_0#1",
  "koulutustyyppi": "yo",
  "metadata": {
    "koulutusalaKoodiUrit": [
      "kansallinenkoulutusluokitus2016koulutusalataso2_1#1"
    ],
    "koulutustyyppi": "yo",
    "kuvauksenNimi": {
      "fi": "Fi kuvauksen nimi"
    },
    "kuvaus": {
      "fi": "Fi kuvaus"
    },
    "lisatiedot": [
      {
        "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
        "teksti": {
          "fi": "koulutuksenlisatiedot_0 kuvaus"
        }
      }
    ],
    "opintojenLaajuusKoodiUri": "opintojenlaajuus_1#1",
    "osaamisalaKoodiUri": null,
    "tutkinnonOsat": [],
    "tutkintonimikeKoodiUrit": [
      "tutkintonimikekk_1#1",
      "tutkintonimikekk_2#1"
    ],
    "tyyppi": "yo"
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "koulutus_0"
  },
  "oid": "1.2.3.4.5.6",
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "1.1.1.1.1.1",
    "1.2.1.1.1.1"
  ],
  "teemakuva": null,
  "tila": "tallennettu"
};

exports[`editKoulutusForm > should be able to edit lukiokoulutus #0`] =
{
  "ePerusteId": null,
  "esikatselu": true,
  "johtaaTutkintoon": true,
  "julkinen": true,
  "kielivalinta": [
    "fi"
  ],
  "koulutusKoodiUri": "koulutus_0#1",
  "koulutustyyppi": "lk",
  "metadata": {
    "koulutusalaKoodiUrit": [],
    "koulutustyyppi": "lk",
    "kuvauksenNimi": {},
    "kuvaus": {},
    "lisatiedot": [
      {
        "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
        "teksti": {
          "fi": "koulutuksenlisatiedot_0 kuvaus"
        }
      }
    ],
    "opintojenLaajuusKoodiUri": null,
    "osaamisalaKoodiUri": null,
    "tutkinnonOsat": [],
    "tutkintonimikeKoodiUrit": [],
    "tyyppi": "lk"
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "koulutus_0"
  },
  "oid": "1.2.3.4.5.6",
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "1.1.1.1.1.1",
    "1.2.1.1.1.1"
  ],
  "teemakuva": null,
  "tila": "tallennettu"
};
